transactionoutbox-core/src/main/java/com/gruelbox/transactionoutbox/DefaultPersistor.java

```diff
@@ -241,14 +245,53 @@ public boolean unblock(Transaction tx, String entryId) throws Exception {
                     + ALL_FIELDS
                     + " FROM "
                     + tableName
-                    + " WHERE nextAttemptTime < ? AND blocked = false AND processed = false LIMIT ?"
+                    + " WHERE nextAttemptTime < ? AND blocked = false AND processed = false and groupid is null LIMIT ?"
                     + forUpdate)) {
       stmt.setTimestamp(1, Timestamp.from(now));
       stmt.setInt(2, batchSize);
       return gatherResults(batchSize, stmt);
     }
   }
 
+  @Override
+  public List<TransactionOutboxEntry> selectBatchOrdered(
+      final Transaction tx, final int batchSize, final Instant now) throws Exception {
+    String forUpdate = dialect.isSupportsSkipLock() ? " FOR UPDATE SKIP LOCKED" : "";
+    try (PreparedStatement stmt =
+        tx.connection()
+            .prepareStatement(
+                dialect.isSupportsWindowFunctions()
+                    ? "WITH t AS"
```

Comment:
Have you done an `EXPLAIN` on these queries on the different DBs and determined if we have suitable indexes?

I have a feeling not having `groupId` anywhere is going to make this pretty slow.

Reply:
I kind of disagree about `groupId` because here we fetch all groups at once so the cardinality of this column is effectively 50/50 (filled or not), and I'm not sure will RDBMS ever use that index even after collecting proper statistics.

At the same time, I think we can add an index for `processed` (it's technically 50/50 too but obviously we'll always have much fewer unprocessed rows that processed ones while selecting, and also hints/statistics can help here). 
We actually have a composite index for `processed, blacklisted, nextAttemptTime` already which should work for a single `processed` column in a `where` clause too (at least in PG, need to check it it other supported DBs)